src/HealthGPS/risk_factor_adjustable_model.cpp

```diff
@@ -0,0 +1,170 @@
+#include "HealthGPS.Core/exception.h"
+#include "HealthGPS.Core/thread_util.h"
+
+#include "risk_factor_adjustable_model.h"
+#include "sync_message.h"
+
+namespace { // anonymous namespace
+
+/// @brief Defines the risk factors adjustment synchronisation message
+using RiskFactorAdjustmentMessage = hgps::SyncDataMessage<hgps::RiskFactorSexAgeTable>;
+
+/// @brief Defines the first statistical moment type
+struct FirstMoment {
+  public:
+    void append(double value) noexcept {
+        count_++;
+        sum_ += value;
+    }
+
+    double mean() const noexcept { return (count_ > 0) ? (sum_ / count_) : 0.0; }
```

Comment:
Throwing an exception would break, since there many not be a person of a particular sex and age in a population, thus `0.0 / 0`. NaN is preferable in this case. One could even just return `sum_ / count_`.

Reply:
... On second thought, no. What if there *is not* a person of a given sex and age in the *baseline* scenario, but there *is* a person of that sex and age in the *intervention* scenario? That person's otherwise valid risk factor would have NaN added to it.

Reply:
Ah ok. Maybe add a comment saying that then?